scripts/generate_usb_vif/generate_vif.py

```diff
@@ -179,72 +151,69 @@ def parse_and_add_sink_pdo_to_xml(xml_ele, pdo_value, pdos_info):
-            new_xml_ele = add_element_to_xml(xml_ele, prop.name)
-            parse_and_add_node_to_xml(new_xml_ele, member)
-        elif isinstance(member, edtlib.ControllerAndData):
-            new_xml_ele = add_element_to_xml(xml_ele, prop.name)
-            parse_and_add_controller_and_data_to_xml(new_xml_ele, member)
-        else:
-            ValueError(
-                f'Noticed undefined type : {str(type(member))}, with value {str(member)}')
-
-
-def parse_and_add_node_to_xml(xml_ele, node):
-    if not isinstance(node, edtlib.Node):
-        return
-    xml_ele = add_element_to_xml(xml_ele, node.name)
-    for prop in node.props:
-        if is_simple_datatype(node.props[prop].val):
-            add_element_to_xml(xml_ele, node.props[prop].name,
-                               str(node.props[prop].val))
-        elif node.props[prop].name == constants.SINK_PDOS:
-            parse_and_add_sink_pdos_to_xml(xml_ele, node.props[prop].val)
-        elif isinstance(node.props[prop].val, list):
-            parse_and_add_array_to_xml(xml_ele, node.props[prop])
-        elif isinstance(node.props[prop].val, edtlib.Node):
-            new_xml_ele = add_element_to_xml(xml_ele, node.props[prop].name)
-            parse_and_add_node_to_xml(new_xml_ele, node.props[prop].val)
-        elif isinstance(node.props[prop].val, edtlib.ControllerAndData):
-            new_xml_ele = add_element_to_xml(xml_ele, node.props[prop].name)
-            parse_and_add_controller_and_data_to_xml(new_xml_ele, node.props[prop].val)
-        else:
-            ValueError(
-                f'Noticed undefined type : {str(type(node.props[prop].val))}, '
-                f'with value {str(node.props[prop].val)}')
-    for child in node.children:
-        new_xml_ele = add_element_to_xml(xml_ele, child)
-        parse_and_add_node_to_xml(new_xml_ele, node.children[child])
+def parse_and_add_sink_pdos_to_xml(xml_ele, sink_pdos):
+    new_xml_ele = add_element_to_xml(xml_ele, dt_constants.SINK_PDOS)
+    snk_max_power = 0
+    pdo_info = dict()
+    for pdo_value in sink_pdos:
+        power_mv = parse_and_add_sink_pdo_to_xml(new_xml_ele, pdo_value,
+                                                 pdo_info)
+        if power_mv > snk_max_power:
+            snk_max_power = power_mv
+    add_element_to_xml(xml_ele, vif_element_constants.PD_POWER_AS_SINK,
+                       f'{snk_max_power} mW',
+                       {other_constants.VALUE: str(snk_max_power)})
+    add_element_to_xml(xml_ele, vif_element_constants.HIGHER_CAPABILITY_SET,
+                       None, {
+                           other_constants.VALUE: get_xml_bool_value(
+                               vif_element_constants.HIGHER_CAPABILITY_SET in pdo_info)})
+    add_element_to_xml(xml_ele, vif_element_constants.NUM_SNK_PDOS, None,
+                       {other_constants.VALUE: str(len(sink_pdos))})
+
+
+def parse_and_add_component_to_xml(xml_ele, node):
```

Comment:
Nit: IMO, these long, highly nested functions could use some more blank lines than they have now.

Reply:
Ok, I actually wrote them in one line, when I formatted it in IDE, it broke it that way. Will change it

Reply:
For example, this function is 20 LOC with 5 levels of indentation. I think it would be easier to follow if there were blank lines before lines 200, 213, and maybe 207.

Similarly, `parse_and_add_sink_pdo_to_xml` is 78 lines long with 4 levels of indentation. I think it would be easier to read with blank lines before lines 101, 107, 123, 140, 160, and 171.

Reply:
Added few lines in new revision